Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -864,4 +866,119 @@ ErrorOr<GC::Ref<Key>> convert_a_value_to_a_multi_entry_key(JS::Realm& realm, JS:
+
+            // 6. Assert: status is true.
+            VERIFY(status);
+
+            // 7. Increase i by 1.
+            i++;
+        }
+
+        // 4. Return result.
+        return result;
+    }
+
+    auto key_path_string = key_path.get<String>();
+
+    // 2. If keyPath is the empty string, return value and skip the remaining steps.
+    if (key_path_string.is_empty())
+        return value;
+
+    // 3. Let identifiers be the result of strictly splitting keyPath on U+002E FULL STOP characters (.).
+    auto identifiers = MUST(key_path_string.split('.'));
+
+    // 4. For each identifier of identifiers, jump to the appropriate step below:
+
+    for (auto const& identifier : identifiers) {
+        // If Type(value) is String, and identifier is "length"
+        if (value.is_string() && identifier == "length") {
+            // Let value be a Number equal to the number of elements in value.
+            value = JS::Value(value.as_string().utf16_string_view().length_in_code_units());
+        }
+
+        // If value is an Array and identifier is "length"
+        else if (value.is_object() && is<JS::Array>(value.as_object()) && identifier == "length") {
+            // Let value be ! ToLength(! Get(value, "length")).
+            value = JS::Value(MUST(length_of_array_like(realm.vm(), value.as_object())));
+        }
+
+        // If value is a Blob and identifier is "size"
+        else if (value.is_object() && is<FileAPI::Blob>(value.as_object()) && identifier == "size") {
+            // Let value be value’s size.
+            value = JS::Value(static_cast<FileAPI::Blob&>(value.as_object()).size());
+        }
+
+        // If value is a Blob and identifier is "type"
+        else if (value.is_object() && is<FileAPI::Blob>(value.as_object()) && identifier == "type") {
+            // Let value be a String equal to value’s type.
+            value = JS::PrimitiveString::create(realm.vm(), static_cast<FileAPI::Blob&>(value.as_object()).type());
+        }
+
+        // If value is a File and identifier is "name"
+        else if (value.is_object() && is<FileAPI::File>(value.as_object()) && identifier == "name") {
+            // Let value be a String equal to value’s name.
+            value = JS::PrimitiveString::create(realm.vm(), static_cast<FileAPI::File&>(value.as_object()).name());
+        }
+
+        // If value is a File and identifier is "lastModified"
+        else if (value.is_object() && is<FileAPI::File>(value.as_object()) && identifier == "lastModified") {
+            // Let value be a Number equal to value’s lastModified.
+            value = JS::Value(static_cast<double>(static_cast<FileAPI::File&>(value.as_object()).last_modified()));
+        }
+
+        // Otherwise
+        else {
+            // 1. If Type(value) is not Object, return failure.
+            if (!value.is_object())
+                return Error::from_string_literal("Value is not an object");
```

Comment:
Feels weird to return just an AK::Error with a string view here. Can we not return a `vm.throw_completion<JS::TypeError>(JS::ErrorType::NotAnObject, value);` for this? I guess that would need a spec issue...

Reply:
`AK::Error` is used as the `failure` type here, its not an error in the sense of a normal error, but as a sentinel value

Reply:
Eh. I'm not a big fan of the sentinel value concept. I'd rather the spec just state what kind of JS Exception or WebIDL DOMException or w/e to return. But we can keep it like this for now.